### PR TITLE
Bootloader force user input

### PR DESCRIPTION
**Pull request by jreidinger**
## Problem

In order to support the QA use-case, we need to make it possible to configure the boot loader to stop at the menu without a timeout.

trello: https://trello.com/c/dKYmc0K7/4018-5-agama-ped-10810-new-section-to-configure-bootloader
jira:
- [PED-10810](https://jira.suse.com/browse/PED-10810)
- Epic: https://jira.suse.com/browse/AGM-54
- Story: https://jira.suse.com/browse/AGM-59
- Tasks: https://jira.suse.com/browse/AGM-67 and https://jira.suse.com/browse/AGM-68

## Solution

Add to profile section `Bootloader` with key `stopOnBootMenu` and boolean value. This allows QA to use `jq -n ".bootloader.stopOnBootMenu = true" | agama config load` after selecting product and before start of installation to force stop without any timeout.

## Testing

- *Added a new unit test*
- *Tested manually*

### CODE REVIEW

File: rust/agama-lib/src/bootloader/client/bootloader.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) [2024] SUSE LLC
```
Comment (imobachgs) [inline rust/agama-lib/src/bootloader/client/bootloader.rs:1]: For consistency, I would follow the same approach as the other modules: `http_client.rs` and `client.rs` inside of `bootloader/` (or `http_client.rs` and `dbus_client.rs`).

File: rust/agama-lib/src/bootloader/model/bootloader.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) [2024] SUSE LLC
```
Comment (imobachgs) [inline rust/agama-lib/src/bootloader/model/bootloader.rs:1]: Why `bootloader::model::bootloader`? If the model is simple enough, I would go for just `bootloader::model`.

**Review by imobachgs — COMMENTED** [review]
The overall approach looks good. Thanks!

File: rust/agama-lib/src/bootloader/client/bootloader.rs (hunk 1 above)
Comment (jreidinger) [reply rust/agama-lib/src/bootloader/client/bootloader.rs:1]: yeah, this was copy from previous location under storage directory. Will fix.

File: rust/agama-lib/src/bootloader/model/bootloader.rs (hunk 2 above)
Comment (jreidinger) [reply rust/agama-lib/src/bootloader/model/bootloader.rs:1]: agreed. Same explanation as above.

File: rust/agama-lib/src/bootloader/model/bootloader.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+//! Implements a data model for Bootloader configuration.
+
+use serde::{Serialize, Deserialize};
+
+/// Represents a Bootloader
+#[derive(Clone, Debug, Serialize, Deserialize, Default, utoipa::ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct Bootloader {
+    pub force_user_input: bool,
```
[the remark this answers is not in the document]
Comment (jreidinger) [reply rust/agama-lib/src/bootloader/model/bootloader.rs:29]: I am also not sure about ideal name...For me advantage of `force_user_input` over `stop_on_boot` is that it emphasize what user have to do ( press enter or something else ).

File: rust/agama-lib/src/bootloader.rs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+//! Implements support for handling the storage settings
+
+pub mod client;
+// pub mod http_client;
```
Comment (imobachgs) [inline rust/agama-lib/src/bootloader.rs:24]: leftover?

**Review by imobachgs — APPROVED** [review]
It looks good. Thanks a lot for helping to unblock QA 🥳

**Review by imobachgs — APPROVED** [review]

**Pull request merged**